src/home_robot_hw/home_robot_hw/env/stretch_image_nav_env.py

```diff
@@ -0,0 +1,207 @@
+from typing import Any, Dict, Optional
+
+import numpy as np
+import cv2
+from omegaconf import DictConfig
+
+import rospy
+
+import home_robot
+from home_robot.motion.stretch import STRETCH_HOME_Q
+from home_robot.core.interfaces import Action, DiscreteNavigationAction, Observations
+from home_robot.utils.geometry import xyt2sophus, xyt_base_to_global
+from home_robot_hw.env.stretch_abstract_env import StretchEnv
+
+
+class StretchImageNavEnv(StretchEnv):
+    """Create a detic-based object nav environment"""
+
+    def __init__(self, config: Optional[DictConfig] = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if config:
+            self.forward_step = config.habitat.simulator.forward_step_size  # in meters
+            self.rotate_step = np.radians(config.habitat.simulator.turn_angle)
+            self.image_goal = self._load_image_goal(config.stretch_goal_image_path)
+        else:
+            self.forward_step = 0.25
+            self.rotate_step = np.radians(30)
+            self.image_goal = None
+        self.reset()
+
+    def _load_image_goal(self, goal_img_path) -> np.ndarray:
+        goal_image = cv2.imread(goal_img_path)
+        # opencv loads as BGR, but we use RGB.
+        goal_image = goal_image[:, :, ::-1]
+        assert goal_image.shape[0] == 512
+        assert goal_image.shape[1] == 512
+        return goal_image
+
+    def reset(self):
+        self._episode_start_pose = xyt2sophus(self.get_base_pose())
+        self.goto(STRETCH_HOME_Q)
+
+    def apply_action(self, action: Action):
```

Comment:
docstring explaining what action means, how it should be formatted would be great.